### ISSUE TITLE
 CoL of November 2022

### ISSUE THREAD

**Issue by yroskov** [CatalogueOfLife/testing#211]
- [x] WoRMS of 2022-11-01; imported 2022-11-03 (-04 Mollusca); synced 2022-11-04; re-imported 2022-11-09 (-10 Mollusca) - after https://github.com/CatalogueOfLife/data/issues/483#issuecomment-1305456792; 
Problem, 2022-11-11: broken sectors in WoRMS Brachyura https://github.com/CatalogueOfLife/testing/issues/39#issuecomment-1311739663

- [x] ITIS of 2022-10-28; imported 2022-11-03; new sector mite family Melicharidae; draft sync 2022-11-09; re-synced 2022-11-11; **re-synced 2022-11-17**

- [x] Scarabs of 2022-11-08; imported 2022-11-09; **synced 2022-11-10**

- [x] SF Coreoidea of 2022-09-05; **synced 2022-10-26**
- [x] SF Lygaeoidea of 2022-09-05; **synced 2022-10-26**
- [x] SF Mantodea of 2022-09-05; **synced 2022-10-26**
- [x] SF Phasmida of 2022-09-05; **synced 2022-10-27**
- [x] SF Isoptera of 2022-10-20; **synced 2022-10-27**
- [x] SF Dermaptera of 2022-09-05; **synced 2022-10-31**
- [x] SF Embioptera of 2022-09-05; synced 2022-10-31; **re-synced 2022-11-09**
- [x] SF Grylloblattodea of 2022-09-05; **synced 2022-10-31**
- [x] SF Mantophasmatodea of 2022-09-05; **synced 2022-11-01**
- [x] SF Plecoptera **synced 2022-11-02**
- [x] SF Psocodea **synced 2022-11-03**
- [x] SF Zoraptera **synced 2022-11-03**

- [x] StaphBase - need to fix missing species (**{incertae sedis}** and **(incertae sedis)** in Tribe and Subtribe) = FIXED. 
Add family Histeridae as replacement for ITIS Histeridae = completed, draft sync 2022-11-03. 
New problem: inaccessible TASKS reports
**Synced 2022-11-17** with unresolved TASKS

- [x] 3i Auchenorrhyncha - @proceps reminded to fix [24] subspecies without parent species in the root of the Tree https://www.checklistbank.org/dataset/2317/classification
and
https://www.checklistbank.org/dataset/2317/names?facet=rank&facet=issue&facet=status&facet=nomStatus&facet=nameType&facet=field&facet=authorship&facet=authorshipYear&facet=extinct&facet=environment&facet=origin&issue=parent%20id%20invalid&limit=50&offset=0
= FIXED; re-synced 2022-11-16; **re-synced 2022-11-28**

- [x] Systema Dipterorum, ver. 3.10 of 2022-09-30; received 2022-10-02; imported 2022-10-06; **incorrect version and date in metadata - cannot fix.**=FIXED 2022-10-19, metadata updated; synced 2022-10-25 with unresolved genera in square brackets = unable to fix; synced 2022-11-14; **re-synced 2022-11-16** (perhaps with many "misplaced" genera)  

- [x] World Ferns, ver 14.3, Nov 2022, received 2022-10-15, new **ver 14.2, Oct 2022, received 2022-11-08**; imported 2022-11-09; synced 2022-11-10; **re-synced 2022-11-14**

- [x] World Plants, ver 14.3, Nov 2022, received 2022-10-15, new **ver 14.2, Oct 2022, received 2022-11-08**; imported 2022-11-10; synced  **2022-11-16**

- [ ] Species Fungorum Plus - Paul>"For the last few weeks I have been updating Species Fungorum and resolving some internal inconsistencies / errors in preparation for another download for COL+, partly prompted by a request from the GNI for a new download. I may have to get back to you if I forget how to ‘format’ the data, but this may be in early January.
2022-01-25 >I should be in a position to start preparing the download in the middle of May
2022-06-23 >If I do not complete the task by this time next week (last day of June; just over 500 left) I will just ‘hide’ the names from access via the web site and internally tag them as not yet fully looked for.
2022-10-26 >I have recently set a deadline for completion of all the new stuff by the end of the month and then November will be dedicated to resolving a few inconsistencies in Species Fungorum and initiating the download.

- [ ] WCVP of 2022-10-27. Markus 2022-11-09: WCVP with distribution will be available soon in CLB
Update 2022-11-10: WCVP is imported into the main dataset: https://www.checklistbank.org/dataset/2232/imports
Just expect the metadata to slightly change, otherwise ready to use for COL
imported 2022-11-09; synced by mistake 2022-11-18 as a part of GSDs list with re-established Codes = this caused corruption of plant classification in the PREVIEW of 2022-11-18.
re-synced with resolved sectors and unresolved Issues & Tasks 2022-11-28; families Euphorbiaceae, Rubiaceae, Sapotaceae, Lamiaceae re-synced 2022-11-29 with resolved _ACC-ACC sp DiffAuth_ in CoL families.

- [ ] WWW of 2022-08-17

- [ ] Mosses for December(?): https://www.checklistbank.org/dataset/170394/about.  
Passed to TG https://github.com/CatalogueOfLife/data/issues/416#issuecomment-1318726256

**Other 11 GSDs for updates:** https://docs.google.com/spreadsheets/d/1WRxfYQF8h2Xiu3gNUvQ3de-zRH-BSq3wYWIAbHHBrhY/edit?usp=sharing

**Comment by yroskov** [comment]
Preview release (with SFs & WoRMS) started 2022-11-06

**Comment by yroskov** [comment]
Preview release (with fix in SF Embioptera, plus ITIS) started 2022-11-09.

**Comment by yroskov** [comment]
Preview release (with fixed (?) prov acc genera in Systema Dipterorum, Scarabs & WFerns) started 2022-11-14.

**Comment by yroskov** [comment]
Synced 2022-11-18:
WTaxa (Abantiades, Abaris, Absonus - to check whether "prov acc" status via TASKS-project works)

as well as GSDs with assigned Code:
ITIS
Gelechiidae
Brassicaceae
WCVP
WCVP Fabaceae
Global Lepidoptera Index

**Comment by yroskov** [comment]
Preview release started 2022-11-18

**Comment by yroskov** [comment]
Checks of PREVIEW 2022-11-18 (continue)

- [ ] WCVP: unexpected loss of 103K spp after sync of 2022-11-18. 

**There are many disappeared sectors in Asssembly**, which are not reported in Project - Sectors as broken. (about 10 broken sectors in the report have been rematched from the report. Only rematch for family Xeronemataceae & Ginkgo biloba failed; re-established in Assembly).
As further investigation shown, total mess in CoL classification happened after a sync of WCVP. WCVP, World Plants & World Ferns sectors corrupted.

_Example_ from PREVIEW:

![image](https://user-images.githubusercontent.com/28574252/203128249-73628947-c112-4d60-a222-fe1fab7e6691.png)

As children should be and were before 2022-11-18:
 
![image](https://user-images.githubusercontent.com/28574252/203128455-af79839f-d9bd-48b4-bfaa-788bcc49578e.png)

**Comment by mdoering** [comment]
> * [ ]  WTaxa genera **Abantiades** Broun, T., 1914, **Abaris** Voss, E., 1958 & WoRMS genus **Absonus** Rubio & Rolán, 2021 have been marked as "**provisionally accepted**" via TASKS-project tool; GSDs synced (see MD ticket [Redundant, accepted higher taxa data#463](https://github.com/CatalogueOfLife/data/issues/463)). = Seems, TASKS-project tool does not work (for attention of @mdoering, @thomasstjerne, @gdower). Impossible to resolve duplicated unimonial taxa at this moment.


There does not seem to be any decision recorded: 
https://www.checklistbank.org/catalogue/3/dataset/1039/workbench?facet=rank&facet=issue&facet=status&facet=nomStatus&facet=nameType&facet=field&facet=authorship&facet=authorshipYear&facet=extinct&facet=environment&facet=origin&limit=50&offset=0&q=Abantiades

Ah, I can see why. When I use the project duplicates tool this is the request send:

`{"datasetKey":3,"subjectDatasetKey":1039,"subject":{"id":"5b4b8620-6c56-4c2c-ab54-f90f3789a102","parent":"Animalia","name":"Abaris","authorship":"Voss, E., 1958","rank":"genus","status":"accepted"},"mode":"update","status":"provisionally accepted"}
`

The dataset keys are correct, but the subject ID corresponds to the project usage, not the ID in the subject source!
@thomasstjerne the source id can be fetched via http://api.checklistbank.org/dataset/3/nameusage/5b4b8620-6c56-4c2c-ab54-f90f3789a102/source

**Comment by mdoering** [comment]
> **There are many disappeared sectors in Asssembly**, which are not reported in Project - Sectors as broken. (about 10 broken sectors in the report have been rematched from the report. Only rematch for family Xeronemataceae & Ginkgo biloba failed; re-established in Assembly).
> As further investigation shown, total mess in CoL classification happened after a sync of WCVP. WCVP, World Plants & World Ferns sectors corrupted.

I think I know what is happening. There are no dedicated family records in WCVP, but these get created by the importer as "denormalised" records because the classification was given flat for every record. Therefore the families also do not come with any ID, but the importer assigns a unique, sequential identifier. This version of WCVP is quite different and the same families therefore receive different sequential IDs. But because these are sequential they did exist in the past already - but identified different families then. E.g. the former ID `x35S` was used for the Alismatales family [Scheuchzeriaceae](http://api.checklistbank.org/dataset/3/sector/1608). The same id now identifiers [Equisetaceae](https://www.checklistbank.org/dataset/2232/taxon/x35S), so the sector is not broken (the id still exists) and places the wrong family Equisetaceae under Alismatales.

I have just rematched the Scheuchzeriaceae sector which worked and now assigned the current id [xJ6](https://www.checklistbank.org/dataset/2232/taxon/xJ6) for Scheuchzeriaceae to the [sector](http://api.checklistbank.org/dataset/3/sector/1608).

So basically all you need to do is rematch all WCVP sectors and do a resync of all those sectors afterwards.

**Comment by mdoering** [comment]
We should think if we can automate this so the rematching is done automatically for all sectors for some sources that are known to have unstable ids.

**Comment by yroskov** [comment]
>So basically all you need to do is rematch all WCVP sectors and do a resync of all those sectors afterwards.

_Experiment:_ Alismatales families rematched and resynced. Result: 12 families of 14 are OK, but Posidoniaceae is missing in Alismatales and Tapisciaceae from Huerteales incorrectly placed here.

![image](https://user-images.githubusercontent.com/28574252/203642088-6b328d86-3466-4f4d-8b78-4365ed259d68.png)

Seems, proposed way does not guarantee correct repair.

**Comment by mdoering** [comment]
Posidoniaceae shows as broken now, together with 3 other sectors incl Ruppiaceae for Alismatales: https://www.checklistbank.org/catalogue/3/sector?broken=true&limit=100&offset=0&subjectDatasetKey=2232

Yes, you will need to fix those broken sectors manually. Normal procedure for a major upgrade of a source dataset I would think.

Tapisciaceae is a sector from World Plants and that still exists. The duplicate in Alismatales is likely some other wrongly matched subject ID I would not worry about until all sectors are properly rematched.

**Comment by mdoering** [comment]
@thomasstjerne is there a way to manually fix sector subjects now on prod?

**Comment by thomasstjerne** [comment]
In prod you have to go through the tree https://www.checklistbank.org/catalogue/3/assembly?assemblyTaxonKey=6fee264d-ef68-46b8-ac00-65e76ce7b293
If you right click the sector, there is an option "edit sector". This opens a form that allows editing.

**Comment by mdoering** [comment]
The search/suggest does not find any Gingko record although it is there:

https://www.checklistbank.org/dataset/2232/taxon/334053
https://api.checklistbank.org/dataset/2232/nameusage/suggest?q=Gingko
https://api.checklistbank.org/dataset/2232/nameusage/search?q=Gingko

The dataset also appears to be fully indexed. I'll reindex nevertheless

**Comment by mdoering** [comment]
The `Posidoniaceae` rematching error response is this:

`Sector Sector{1605, datasetKey=3, mode=ATTACH, subjectDatasetKey=2232, subject=ACCEPTED FAMILY Posidoniaceae} seems to be a duplicate of Sector{1715, datasetKey=3, mode=ATTACH, subjectDatasetKey=2232, subject=ACCEPTED FAMILY Opiliaceae [x32R]} for subject Posidoniaceae in project 3. Keep sector 1605 broken`

https://api.checklistbank.org/dataset/3/sector/1605
https://api.checklistbank.org/dataset/3/sector/1715

The taxonID [x32R](https://api.checklistbank.org/dataset/2232/taxon/x32R) is Posidoniaceae and it is currently been (wrongly) used for Opiliaceae which hasnt been rematched. Having 2 sectors with the same subjectID is forbidden, so the rematch fails. When all sectors are rematched this should be solved. I am trying that now...

**Comment by mdoering** [comment]
Rematching all of WCVP results in:

<img width="708" alt="image" src="https://user-images.githubusercontent.com/327505/203789181-a89cadd0-7a3b-4695-83c6-fc5bc0c9b143.png">

Rather good. Leaves a few broken ones which could be rematched individually quickly, leaving just `G.biloba` and [Xeronemataceae](https://www.checklistbank.org/dataset/2232/names?q=Xeronemataceae) which has this log:

`Sector Sector{1626, datasetKey=3, mode=ATTACH, subjectDatasetKey=2232, subject=ACCEPTED FAMILY Xeronemataceae} seems to be a duplicate of Sector{1794, datasetKey=3, mode=ATTACH, subjectDatasetKey=2232, subject=ACCEPTED FAMILY Xeronemataceae [x32M]} for subject Xeronemataceae in project 3. Keep sector 1626 broken
`

There are indeed 2 sectors for the same family!
https://api.checklistbank.org/dataset/3/sector/1626
https://api.checklistbank.org/dataset/3/sector/1794

The later one being created today by @yroskov. Please DO NOT MANUALLY CREATE/DELETE sectors - this cause extra troubles in solving this. It also blurs any sector history. Deleting sectors should only be done when the sector indeed should not exist, NEVER to repair broken sectors!!!

**Comment by mdoering** [comment]
I have deleted the news sector and rematched the old one. All sectors but (G.biloba) are matched properly now and can be resynced - which I have started just now too.

**Comment by mdoering** [comment]
syncs are all through. Alismatales looking good:

<img width="359" alt="image" src="https://user-images.githubusercontent.com/327505/203824094-b764add9-7d9e-4400-9e6e-cc09f27a7d18.png">

**Comment by yroskov** [comment]
2022-11-28, Donald:
- [ ] Global Lepidoptera Index, Gelechiidae and Pterophoroidea

**Comment by yroskov** [comment]
Preview release started 2022-11-29
